test/diag2dwd.jl

```diff
@@ -217,6 +218,317 @@ end
     @test nparts(advdiffdp, :Op2) == 2
 end
 
+@testset "Decapode Composition" begin
+
+  """
+      function unique_by!(acset, column_names::Vector{Symbol})
+
+  Given column names from the same table, remove duplicate rows.
+  
+  WARNING: This function does not check if other tables index into the one
+  given. Removal of rows is performed with prejudice.
+
+  See also: [`unique_by`](@ref).
+
+  # Examples
+  ```julia-repl
+  julia> g = @acset Graph begin
+    V = 2
+    E = 3
+    src = [1,1,2]
+    tgt = [1,1,2]
+  end
+  julia> unique_by!(g, :E, [:src, :tgt])
+  ``` 
+  """
+  function unique_by!(acset, table::Symbol, columns::Vector{Symbol})
+    rows = columns .|> (x -> acset[x]) |> (x -> zip(x...)) |> collect
+    rem_parts!(acset, table,
+      setdiff(parts(acset, table),
```

Comment:
setdiff is generally not category theoretic as an operation. I think that `unique_by` would be better expressed as computing the projection onto a quotient by an equivalence relation, rather than an imperative procedure of deleting rows.

The equivalence relation is that two objects in the C-Set are equivalent iff they have the same image for some tuple of homs. There should be a more CT friendly way of capturing that.